huaweicloud/services/nat/resource_huaweicloud_nat_snat_rule.go

```diff
@@ -0,0 +1,326 @@
+package nat
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/nat/v2/snats"
+	"github.com/chnsz/golangsdk/openstack/networking/v1/eips"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+type SourceType int
+
+const (
+	SourceTypeVpc SourceType = 0
+	SourceTypeDc  SourceType = 1
+)
+
+func ResourceSnatRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSnatRuleCreate,
+		ReadContext:   resourceSnatRuleRead,
+		UpdateContext: resourceSnatRuleUpdate,
+		DeleteContext: resourceSnatRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the SNAT rule is located.",
+			},
+			"floating_ip_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: utils.SuppressSnatFiplistDiffs,
+				Description:      "The IDs of floating IPs connected by SNAT rule.",
+			},
+			"gateway_id": {
```

Comment:
is it necessary to replace nat_gateway_id with gateway_id?
if so, add some limit to avoid that they are both empty

Reply:
The prefix `nat` is unnecessary.
I add a check to the function `buildPublicSnatRuleCreateOpts`.